src/runtime/operator.cs

```diff
@@ -0,0 +1,125 @@
+            // TODO: Rich compare, inplace operator support
+            OpMethodMap = new Dictionary<string, Tuple<string, int>>
+            {
+                ["op_Addition"] = Tuple.Create("__add__", TypeOffset.nb_add),
+                ["op_Subtraction"] = Tuple.Create("__sub__", TypeOffset.nb_subtract),
+                ["op_Multiply"] = Tuple.Create("__mul__", TypeOffset.nb_multiply),
+#if PYTHON2
+                ["op_Division"] = Tuple.Create("__div__", TypeOffset.nb_divide),
+#else
+                ["op_Division"] = Tuple.Create("__truediv__", TypeOffset.nb_true_divide),
+#endif
+                ["op_BitwiseAnd"] = Tuple.Create("__and__", TypeOffset.nb_and),
+                ["op_BitwiseOr"] = Tuple.Create("__or__", TypeOffset.nb_or),
+                ["op_ExclusiveOr"] = Tuple.Create("__xor__", TypeOffset.nb_xor),
+                ["op_LeftShift"] = Tuple.Create("__lshift__", TypeOffset.nb_lshift),
+                ["op_RightShift"] = Tuple.Create("__rshift__", TypeOffset.nb_rshift),
+                ["op_Modulus"] = Tuple.Create("__mod__", TypeOffset.nb_remainder),
+                ["op_OneComplement"] = Tuple.Create("__invert__", TypeOffset.nb_invert)
+            };
+
+            _pyOpNames = new Dictionary<string, string>();
+            foreach (string name in OpMethodMap.Keys)
+            {
+                _pyOpNames.Add(GetPyMethodName(name), name);
+            }
+        }
+
+        public static void Initialize()
+        {
+            _opType = GetOperatorType();
+        }
+
+        public static void Shutdown()
+        {
+            if (_opType != null)
+            {
+                _opType.Dispose();
+                _opType = null;
+            }
+        }
+
+        public static bool IsOperatorMethod(string methodName)
+        {
+            return OpMethodMap.ContainsKey(methodName);
+        }
+
+        public static bool IsPyOperatorMethod(string pyMethodName)
+        {
+            return _pyOpNames.ContainsKey(pyMethodName);
+        }
+
+        public static bool IsOperatorMethod(MethodBase method)
+        {
+            if (!method.IsSpecialName)
+            {
+                return false;
+            }
+            return OpMethodMap.ContainsKey(method.Name);
+        }
+
+        public static void FixupSlots(IntPtr pyType, Type clrType)
+        {
+            IntPtr tp_as_number = Marshal.ReadIntPtr(pyType, TypeOffset.tp_as_number);
+            const BindingFlags flags = BindingFlags.Public | BindingFlags.Static;
+            Debug.Assert(_opType != null);
+            foreach (var method in clrType.GetMethods(flags))
+            {
+                if (!IsOperatorMethod(method))
+                {
+                    continue;
+                }
+                var slotdef = OpMethodMap[method.Name];
+                int offset = slotdef.Item2;
+                IntPtr func = Marshal.ReadIntPtr(_opType.Handle, offset);
+                Marshal.WriteIntPtr(pyType, offset, func);
```

Comment:
Just clarifying, that this takes the default implementations for operator slots from a class, that defines them explicitly, and copies them to the target type?

Reply:
It makes slot defines for Python type, so that we can override the operator in `GetClassInfo`, without this, even you set some operator methods, Python would accept them on operator operation, like `a + b`.

Reply:
Yes, this copies the default implementations for operator slots from a class that defines them explicitly.  As explained in [Python behind the scenes #6: how Python object system works](https://tenthousandmeters.com/blog/python-behind-the-scenes-6-how-python-object-system-works/), the default implementation of the nb_add slot simply calls `__add__` on the type, so we can change `__add__` later and it will be called by the default implementation in the slot.  The code should be updated to have this information in a comment.